src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -293,6 +293,28 @@ public AzureVMDeploymentInfo createDeployment(
             putVariable(tmp, "resourceTag", deploymentRegistrar.getDeploymentTag().get());
             putVariable(tmp, "cloudTag", template.getAzureCloud().getCloudName());
 
+            // add purchase plan for image if needed in reference configuration
+            if (!useCustomImage && StringUtils.isBlank(template.getImageId())) {
+                VirtualMachineImage image = azureClient.virtualMachineImages().getImage(locationName,
+                        template.getImagePublisher(), template.getImageOffer(), template.getImageSku(), template.getImageVersion());
+                if (image != null) {
+                    PurchasePlan plan = image.plan();
+                    if (plan != null) {
```

Comment:
what if plan is null? error out?

Reply:
Null is ok since some image don't need this arguments at all, only a few images need plan information.